### PR TITLE
fix: unify adblocker engines update process

### PR DESCRIPTION
**Pull request by smalluban**
A few minor fixes to the adblocker engines maintenance and update process:

* Avoids reloading the main engine when no engines were updated
* Simplifies the logic in `observe()` pattern, to make clear what when happens
* Protects `updateEngines` from calling the function before the update finishes, as there are multiple factors, which trigger the update

The update process was simplified and moved to one place. It is only triggered by the `updateEngines()` function. Also, the update is done only, if the user passes the onboarding, and then it saves the timestamp.

### CODE REVIEW

File: extension-manifest-v3/src/background/adblocker.js [hunk 1]
```diff
@@ -19,6 +19,7 @@ import { parse } from 'tldts-experimental';
 import Options, { observe, ENGINES, isPaused } from '/store/options.js';
 
 import * as engines from '/utils/engines.js';
+import * as trackerdb from '/utils/trackerdb.js';
```
Comment (chrmod) [inline extension-manifest-v3/src/background/adblocker.js:22]: trackerdb is already imported - please merge import statements

File: extension-manifest-v3/src/background/adblocker.js [hunk 2]
```diff
@@ -70,18 +72,48 @@ async function reloadEngine(enabledEngines) {
...
+        if (id === engines.CUSTOM_ENGINE) continue;
+        updated = (await engines.update(id).catch(() => false)) || updated;
+      }
+
+      // Reload the main engine after all engines are updated
+      if (updated) await reloadMainEngine();
 
+      // Update TrackerDB engine
+      trackerdb.setup.pending && (await trackerdb.setup.pending);
+      await engines.update(engines.TRACKERDB_ENGINE).catch(() => null);
```
Comment (chrmod) [inline extension-manifest-v3/src/background/adblocker.js:106]: shouldn't trackerdb update even if no other engine is enabled?

Comment (smalluban) [reply extension-manifest-v3/src/background/adblocker.js:106]: I made it on purpose - I think after all, it's best to all only onboarded users to update the trackerdb more often than a release - those not onboarded will have an update with a new release. It also simplifies the case by setting the timestamp only when engines are enabled. Without that, it must be set not to update trackerdb repeatedly if the user doesn't want to onboard.

Comment (chrmod) [reply extension-manifest-v3/src/background/adblocker.js:105]: okay, wanted to be sure that's the case

**Review by chrmod — APPROVED** [review]

**Pull request merged**